### ISSUE TITLE
centos7 guest fails to mount smb share from windows host

### ISSUE THREAD

**Issue by billfergus** [geerlingguy/drupal-vm#1147]
### Issue Type

 - Bug Report / Support Request

### Your Environment

```
$ vagrant --version && echo "VirtualBox `vboxmanage --version`" && ansible --version
Vagrant 1.9.0
-bash: vboxmanage: command not found
VirtualBox
-bash: ansible: command not found

About VirtualBox...
Version 5.1.14 r112924 (Qt5.6.2)
```

### Your OS

 - **Windows** (7 Enterprise SP1)

### Full console output

https://gist.github.com/billfergus/4dbb42c1ed84f9e5ed808fb5f35d86d4

### Summary

centos7 guest fails to mount smb share from windows host because linux packages that are required to support smb in the guest are not installed in the geerlingguy/centos7 box.

**Comment by geerlingguy** [comment]
Are you using `type: "smb"` in your configuration in `config.yml`?

See [Vagrant's smb documentation](https://www.vagrantup.com/docs/synced-folders/smb.html); only Vagrant synced folders are supported out of the box (e.g. smb folder in windows synced _into_ the VM, and not vice-versa).

**Comment by billfergus** [comment]
Yes, please see my config.yml and Vagrantfile.local here: https://gist.github.com/billfergus/e6cf3b3ebd325b90fc993d03bffd0c28
I edited the question to more clearly indicate that the smb share is provided by the Windows host, and mounted by the centos7 guest.

The Vagrantfile.local is for adding "vers=2.1" to the options passed to the mount command for both mounts:
- /vagrant
- /var/www/drupalvm
That additional option is required for centos to successfully mount in my environment.

The console output reflects that the mounts work after the cifs-utils package has been installed in the guest and the machine then reloaded. I explored using a pre_provision_script but because the smb mount occurs before provisioning that didn't help, nor did adding cifs-utils to extra_packages.

**Comment by geerlingguy** [comment]
Interesting... I don't think I've tried SMB on a CentOS guest in Windows before. Do you know if it works out of the box on Ubuntu?

**Comment by billfergus** [comment]
Yes, it works out of the box with geerlingguy/ubuntu1604. I tested this by changing vagrant_box in the config.yml file that I supplied above. I also re-initialized things in the environment (deleted smb shares, removed .vagrant.d, installed vagrant plugins, removed and replaced the drupal-vm directory).

It looks like cifs-utils is installed in geerlingguy/ubuntu1604:

```
 ~/desktop/drupal-vm
$ vagrant ssh drupalvm
Welcome to Ubuntu 16.04.1 LTS (GNU/Linux 4.4.0-31-generic x86_64)

 * Documentation:  https://help.ubuntu.com
 * Management:     https://landscape.canonical.com
 * Support:        https://ubuntu.com/advantage
vagrant@drupalvm:~$ apt-get -q install aptitude
```
...

```
vagrant@drupalvm:~$ aptitude show cifs-utils | grep -i state
State: installed
vagrant@drupalvm:~$
```

I left everything other than the vagrant_box change the same, so the "vers=2.1" option was included. I'd need to rebuild again without it to determine if it is necessary for ubuntu as well.

**Comment by geerlingguy** [comment]
@billfergus - Thanks so much for the detail. I may need to add it in the upstream box project, https://github.com/geerlingguy/packer-centos-7

The funny thing is, I don't actually manually install cifs-utils as part of the Ubuntu box build, so I wonder if it's just included by default with the Ubuntu server minimal installation... I'd be willing to add it to my base boxes for CentOS for consistency's sake, so Windows users can safely use SMB regardless of OS.

**Comment by billfergus** [comment]
@geerlingguy I rebuilt the geerlingguy/ubuntu1604 box without the the "vers=2.1" option (and after re-initializing things in the environment as before) and the mount failed before provisioning. I think this verifies that for a linux guest to mount a smb share provided by a windows host, these two things are required:
 - cifs-utils package installed in the guest
 - "vers=" option provided for the mount command. This can be set with the "mount_options" key in a [Vagrantfile.local](https://gist.github.com/billfergus/e6cf3b3ebd325b90fc993d03bffd0c28#file-vagrantfile-local) file.

Choices for the "vers=" option are documented in the manpage for mount.cifs:
```
vagrant@drupalvm:~$ man mount.cifs

...

       vers=
           SMB protocol version. Allowed values are:

               1.0 - The classic CIFS/SMBv1 protocol. This is the default.

               2.0 - The SMBv2.002 protocol. This was initially introduced in Windows
               Vista Service Pack 1, and Windows Server 2008. Note that the initial
               release version of Windows Vista spoke a slightly different dialect (2.000)
               that is not supported.

               2.1 - The SMBv2.1 protocol that was introduced in Microsoft Windows 7 and 
               Windows Server 2008R2.

               3.0 - The SMBv3.0 protocol that was introduced in Microsoft Windows 8 and
               Windows Server 2012.

           Note too that while this option governs the protocol version used, not all
           features of each version are available.
```
If you wouldn't mind adding cifs-utils to the upstream box project the next time that you need to rebuild it, it would help save centos users a step. (and we would appreciate the help!)

**Comment by geerlingguy** [comment]
@billfergus - Thanks so much for documenting this here—can you open an issue upstream for it, and I'll try to get that in next time I build a box version? https://github.com/geerlingguy/packer-centos-7/issues

**Comment by billfergus** [comment]
@geerlingguy Thank you for the assistance. https://github.com/geerlingguy/packer-centos-7/issues/20
